### PR TITLE
support running on pod GitHub runner

### PR DESCRIPTION
**Pull request by yardenasadosa**
ec2metadata will work only if we run on EC2.
They would like to raise an error if there are missing credentials and we are not on EC2 that might have IAM role,
So now we support the case of credentials preconfigured in any way.

### CODE REVIEW

**Comment by antgel** [comment]
Wow, can't believe I'm still subscribed to this repo. Will unsubscribe but just wanted to suggest rebase-squashing all these commits for hopefully obvious reasons e.g. making it easier for @reist to review, not cluttering the git log. All the best to everyone!

**Comment by mirono** [comment]
Yep, I was astonished as well. Someone actually still works there :)

**Comment by ayashjorden** [comment]
Riding on the nostalgic wave, I'd agree with the above, and will also add that the README.md needs some TLC as well. But maybe in a separate PR

File: lib/aws_one_click_staging/aws_warrior.rb [hunk 1]
```diff
@@ -104,12 +104,23 @@ def setup_aws_credentials config
...
       end
+
+      #check if there are some credentials on the machine
+      begin
+        identity = Aws::STS::Client.new().get_caller_identity
+      rescue => e
+        p "no credentials"
+      else
+        p "the credentials are:"
+        p identity
```
Comment (reist) [inline lib/aws_one_click_staging/aws_warrior.rb]: Please drop the else branch and put a more specific class in the rescue if there is previous failed job history to use.

**Review by reist — APPROVED** [review]

**Pull request merged**